### PR TITLE
Pe 1706 as a pe i would like time sensitive unit tests to be atomic regardless of environment

### PR DESCRIPTION
**Pull request by DanielKerri-WDFC**
review before merge

### CODE REVIEW

File: HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs [hunk 1]
```diff
@@ -2,78 +2,147 @@
...
+
+                valueFound = selector(webDriver);
+
+                if (predicate(valueFound))
+                    return valueFound;
+
+                throw new NoSuchElementException($"{valueFound} not found");
             }
-            catch (NoSuchElementException)
+            catch (NoSuchElementException e)
```
Comment (odekhtiarenko) [inline HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs]: Why do we need here this try catch block if we rethrowing another exception?
We can remove try catch and just throw new TimeoutException
in this case we can remove  this line T valueFound = default(T);

**Comment by odekhtiarenko** [comment]
can you merge it in one commit? 
Because 1 commit per 1 file is too much

File: HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs (hunk 1 above)
Comment (DanielKerri-WDFC) [inline HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs]: selector(webDriver) may throw NoSuchElementException if the web element was not found within the wait time

Comment (DanielKerri-WDFC) [inline HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs]: when this happens, we would like to know the "message" produced

Comment (DanielKerri-WDFC) [inline HealthMonitoring.AcceptanceTests/Helpers/Selenium/WebDriverExtensions.cs]: This does not change the fact that the error was effectively a timeout exception and is what we really care about

**Comment by DanielKerri-WDFC** [comment]
This will be merged into one commit

**Pull request merged**